node/core/approval-voting/src/lib.rs

```diff
@@ -689,28 +765,46 @@ async fn run<C>(
 
 				actions
 			}
-			background_request = background_rx.next().fuse() => {
-				if let Some(req) = background_request {
-					handle_background_request(
-						&mut ctx,
-						&mut state,
-						&subsystem.metrics,
-						req,
-					).await?
-				} else {
-					Vec::new()
+			approval_state = currently_checking_set.next().fuse() => {
+				let sender = ctx.sender();
+				if let Some((
+					relay_block_hashes,
+					ApprovalState {
+						validator_index,
+						candidate_hash,
+						candidate_index,
+						status
+					}
+				)) = approval_state {
+					if matches!(status, ApprovalOutcome::Approved) {
+						for block_hash in relay_block_hashes.into_iter() {
```

Comment:
Do I need to issue approval for every relay_parent Hash associated with this candidate_hash or is this incorrect?

Reply:
This is 100% correct.

It's an inefficiency, because technically approvals only need to be issued once per candidate, but our networking code relies on the duplicates.